### PR TITLE
Use `WordBreak` based on the page `role`, inside the OnThisPageNav

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 107193315

## Summary

Refactors the OnThisPageNav to use the page `Role` in order to determine if using the WordBreak is necessary or not.

Article and Collection pages most often dont need the WordBreak

## Dependencies

NA

## Testing

Steps:
1. Use a docc archive that has both code and articles in order to test this out, like the DocC docs

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by dobromir-hristov** [comment]
@swift-ci test

File: tests/unit/components/OnThisPageNav.spec.js [hunk 1]
```diff
@@ -6,13 +6,14 @@
  *
  * See https://swift.org/LICENSE.txt for license information
  * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
-*/
+ */
```
Comment (dobromir-hristov) [inline tests/unit/components/OnThisPageNav.spec.js:9]:
```suggestion
*/
```

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Review by SamLanier — APPROVED** [review]
Design approved.

File: src/components/OnThisPageNav.vue [hunk 2]
```diff
@@ -54,6 +62,13 @@ export default {
       url: buildUrl(`#${item.anchor}`, $route.query),
     })),
     currentPageAnchor: ({ store }) => store.state.currentPageAnchor,
+    wordWrapComponent({ role }) {
+      // Articles and Collection pages normally dont need special wrapping for their titles.
+      if (role === TopicRole.collection || role === TopicRole.article) {
```
Comment (mportiz08) [inline src/components/OnThisPageNav.vue:67]: Would it be better to invert this maybe and only use the `WordBreak` for symbol-like roles? I expect that the more common case would be article-like pages, but I may be wrong.

You might have already looked into it and found that this approach works better though—just curious.

Comment (dobromir-hristov) [reply src/components/OnThisPageNav.vue:67]: I was not sure how many other cases we may have, hence why I did it like this.

If you think the `symbol` role is the only one we may need the break for, then I will refactor it, no prob.

Comment (mportiz08) [reply src/components/OnThisPageNav.vue:67]: I looked into this more extensively today, and I think that approach should generally be fine.

The only troubling case would be when a symbol name is used in an overview subheading, but we'll never have enough context to know that an actual symbol is being referenced in that situation, but it should be fairly uncommon and not an awful experience to not have the improved breaks in that situation.

I went ahead and opened #701 for that approach.

**Comment by dobromir-hristov** [comment]
Closing in favour of #701

**Pull request closed without merging**